plugins/runner/internal/runtimehandler/binary/handler.go

```diff
@@ -0,0 +1,241 @@
+// Copyright © 2024 Cisco Systems, Inc. and its affiliates.
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build linux
+
+package binary
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"sync"
+	"syscall"
+
+	multierror "github.com/hashicorp/go-multierror"
+
+	"github.com/openclarity/vmclarity/plugins/runner/internal/runtimehandler"
+	"github.com/openclarity/vmclarity/plugins/runner/types"
+	"github.com/openclarity/vmclarity/plugins/sdk-go/plugin"
+	"github.com/openclarity/vmclarity/utils/fsutils/containerrootfs"
+)
+
+type binaryRuntimeHandler struct {
+	config types.PluginConfig
+
+	cmd        *exec.Cmd
+	stdoutPipe io.ReadCloser
+	stderrPipe io.ReadCloser
+
+	pluginDir            string
+	inputDirMountPoint   string
+	pluginServerEndpoint string
+	ready                bool
+	imageCleanup         func()
+
+	mu sync.Mutex
+}
+
+func New(ctx context.Context, config types.PluginConfig) (runtimehandler.PluginRuntimeHandler, error) {
+	return &binaryRuntimeHandler{
+		config: config,
+	}, nil
+}
+
+//nolint:cyclop
+func (h *binaryRuntimeHandler) Start(ctx context.Context) error {
+	h.mu.Lock()
+	defer h.mu.Unlock()
+
+	image, cleanup, err := containerrootfs.GetImageWithCleanup(ctx, h.config.ImageName)
+	if err != nil {
+		return fmt.Errorf("unable to get image(%s): %w", h.config.ImageName, err)
+	}
+	h.imageCleanup = cleanup
+
+	home, err := os.UserHomeDir()
+	if err != nil {
+		return fmt.Errorf("unable to determine user's home directory: %w", err)
+	}
+
+	h.pluginDir = filepath.Join(home, ".vmclarity/plugins", h.config.Name, image.Metadata.ID)
```

Comment:
it would be good if we could expose two more options on the plugin config to allow users to
a) specify custom dir if they want the plugin data to be saved/loaded to allow caching via `binary_artifacts_path` which would default to `${HOME}/.vmclarity/plugins` if not specified
b) specify cleanup flag if they want to preserve the data or not between runs `binary_artifacts_keep` to allow reuse when used with flag above. We could also use `binary_artifacts_clean` to default to preserve behaviour when not specified

For example
```yaml
## reruns preserve new and reuse existing artifacts
binary_artifacts_path: /tmp/plugin-artifacts
binary_artifacts_keep: true

## reruns reuse existing artifacts between runs but do not preserve the new artifacts
binary_artifacts_path: /tmp/plugin-artifacts
binary_artifacts_keep: false

## reruns neither reuse nor preserve
# empty binary_artifact configs
```

Reply:
we can do this in a separate PR, added an issue https://github.com/openclarity/vmclarity/issues/1854 to track this